src/main/java/com/google/sps/servlets/DataFromDatabaseServlet.java

```diff
@@ -50,13 +58,17 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         Table.Builder tableBuilder = Table.builder().setName(table);
         tableBuilder.setColumnSchemas(columnSchemas);
         Statement queryStatement = constructQueryStatement(columnSchemas, table);
+
+        queryAudit += queryStatement + "; ";
```

Comment:
Since this is inside a for loop, the number of String concatenation depends on the number tables queried, and it would be potentially large. Strings are immutable, use StringBuilder which is mutable in this case.

Reply:
Done!